cpp_code/include/CrossCat/RandomNumberGenerator.h

```diff
@@ -20,26 +20,24 @@
 #ifndef GUARD_randomnumbergenerator_h
 #define GUARD_randomnumbergenerator_h
 
-#include <boost/random/mersenne_twister.hpp>
-#include <boost/random/uniform_01.hpp>
 #include <ctime>
-//
+#include <stdint.h>
+
 #include "constants.h"
+#include "weakprng.h"
 
-/////////////////////////
-// from runModel_v2.cpp
 class RandomNumberGenerator {
 public:
-    RandomNumberGenerator() : _engine(0), _dist(_engine) {};
-    RandomNumberGenerator(int SEED) : _engine(SEED), _dist(_engine) {};
+    RandomNumberGenerator(int seed = 0) { set_seed(seed); }
     double next();
-    int nexti(int max = MAX_INT);
+    int nexti(int bound = MAX_INT);
+    double stdnormal();
+    double stdgamma(double alpha);
```

Comment:
Pushed a bunch more changes to test all the distributions.  Just used psi tests for distributions with CDFs, since with a reasonable number of samples they are sufficiently powerful to distinguish, e.g., t from normal.  These test samplers against parameters that are not special for the sampler, but admit simple forms of the CDFs which makes self-contained testing convenient.